json-decode.go

```diff
@@ -24,8 +24,8 @@ func (cdc *Codec) decodeReflectJSON(bz []byte, info *TypeInfo, rv reflect.Value,
 		panic("should not happen")
 	}
 
-	disambiguate := rv.Kind() != reflect.Interface && (info.Registered || info.AlwaysDisambiguate)
-	if !disambiguate { // No need for diambiguation, decode as is.
+	// No need for diambiguation, decode as is.
```

Comment:
Could we move this comment either back to where it was or under the if statement?
As it is, it might imply that we always don't disambiguate.

Reply:
Oh duh, of course.